### ISSUE TITLE
ambulante Fälle nicht importieren

### ISSUE THREAD

**Issue by schoenenb** [Optinomic/apps#187]
Könnt ihr einstellen, dass es ambulante Fälle nicht mehr aus dem Pabs holt?
Also nur Abteilung (stay.class) = S
Thema aktuell ist, dass bei allen Patienten, welche Stationär sind, BSCL + HoNOS aktiviert werden.
Allerdings macht es dies nicht nur beim stationären Fall, sondern bei allen aktuell offenen Fällen - unter anderem auch den ambulanten. Diese müssten ja eigentlich nicht drin sein, das war aber glaube ich mal die Lösung für die Fälle, welche im Pabs noch nicht erfasst waren, dass trotzdem Daten eingetragen werden konnten.

**Comment by ottigerb** [comment]
Nur noch ambulante Fälle `(stay.class) = S` einlesen ist durch die Schnittstelle definiert. Somit müsstet Ihr mit PABS die Schnittstelle anpassen lassen. Würde ich nicht empfehlen.

Applikationen werden durch Patientengruppen definiert - von welcher Gruppe sprichst Du - resp. welche Gruppe aktiviert die Apps? 

Es gibt ja auch die http://optinomic.suedhang.ch/client.new/#/patients/group/3 welche auf ` WHERE s.class LIKE 'S%'` schaut. Somit könnte die Gruppe welche die Apps aktiviert mit diesem `WHERE s.class LIKE 'S%'` ergänzt werden. Oder verstehe ich etwas falsch?

**Comment by schoenenb** [comment]
http://optinomic.suedhang.ch/client.new/#/patients/group/18
Das ist die Gruppe, wo die Patienten reingeladen werden. Ich habe ihr heute die Aktivierung zugewiesen, weil mit dem Wechsel bei den QM-Assistentinnen die manuelle Zuteilung zur "Eintritte QuEA"-Gruppe etwas vernachlässigt wurde und bei den QuEA-fällen drum z.T. keine Fragebögen aktiviert waren.

Da ist auch danach sortiert, dass nur stationäre Fälle reingenommen werden `WHERE stay.class LIKE 'S%'.` Allerdings gibt es Patienten, die nebenbei eine offenen ambulanten Fall haben, und in diesem ambulanten Fall werden die Module ebenfalls aktiviert. Meine These ist zumindest, dass die Module bei allen noch nicht abgeschlossenen Fällen aktiviert werden, weil ja der Patient an sich ausgewählt wird, und nicht ein spezifischer Fall von ihm.

Früher waren ja überhaupt nur die stationären und teilstationären Fälle in Optinomic drin, weil die ambulanten nichts auszufüllen haben, drum dachte ich, das könnte man relativ einfach wieder so einstellen...

**Comment by ottigerb** [comment]
Genau:  Der eigentliche Patient wird selektiert und dann erhalten alle offenen/aktive Fälle die Apps.

Allenfalls wird eine STAY-Group relevant:  z.B.:  http://optinomic.suedhang.ch/client.new/#/stays/group/2

**Comment by schoenenb** [comment]
Das wär's wahrscheinlich! Funktionieren die schon? So dass man die nach dem gleichen SQL filtern könnte?

**Comment by ottigerb** [comment]
Funktioniert analog Patientengruppe!  
=> Somit müssten wir über den Fall gehen - es gilt ein SQL-Code zu finden - wie z.B.:

```
SELECT s.* FROM stay AS s
WHERE s.class='S3'
```

**Comment by ottigerb** [comment]
Ist m.E. gelöst - vgl. :  http://optinomic.suedhang.ch/client.new/#/stays/group/7  sowie http://optinomic.suedhang.ch/client.new/#/news/detail/e882089e-5353-47dd-998a-cfa9df07531d


### Patients


```SQL 
SELECT 
p.* FROM patient AS p

LEFT JOIN stay ON(p.id = stay.patient) 

WHERE EXISTS (
    SELECT 1 
    FROM ch_suedhang_bot_pabs_odbc.ch_suedhang_bot_pabs_odbc_pabs_stays AS pabs
    WHERE pabs.org_short = 'PTS' AND stay.cis_fid::varchar = pabs.cis_fid
)
AND stay.cis_fid != 0
AND (stay.stop is null OR stay.stop >= (now() - interval '1 day'))
AND stay.start <= now()
```

### Stays
```SQL 
SELECT 
s.* FROM stay AS s

WHERE EXISTS (
    SELECT 1 
    FROM ch_suedhang_bot_pabs_odbc.ch_suedhang_bot_pabs_odbc_pabs_stays AS pabs
    WHERE pabs.org_short = 'PTS' AND s.cis_fid::varchar = pabs.cis_fid
)
AND s.cis_fid != 0
AND (s.stop is null OR s.stop >= (now() - interval '1 day'))
AND s.start <= now()
```

**Comment by schoenenb** [comment]
Ist es nicht doch möglich, die ambulanten Fälle wegzulassen? Wir haben immer wieder Mühe mit Fragebögen, die im ambulanten Fall mit Aufenthalt 1.1.1970 - unbekannt ausgelöst werden. Diese erscheinen bei den PatientInnen dann im Assessment und schlussendlich muss man alle Antworten mit dem richtigen Fall verknüpfen. Wieso es den 1.1.1970 nimmt und nicht das tatsächlich eingegebene Eintrittsdatum (& Austrittsdatum) weiss ich nicht mehr.

In meiner Vorstellung sähe das so aus: wenn im Schnittstellenfile (laut aktueller Definition der DDAG: PV1, Abschnitt 20 (Patientenklasse)) A für ambulant steht, dass es KEINEN Fall eröffnet, bzw. nur, wenn dort S für stationär steht, dass es dies macht.

**Comment by schoenenb** [comment]
Um das Anliegen noch zu bekräftigen:
In den falschen Fällen eingegebene HoNOS-Daten blockieren den Gesamten Export des Erhebungstages - so geschehen am 5.2. und 27.3., entsprechend konnten auch alle anderen Honösser nicht ins PABS importiert werden.

**Comment by ottigerb** [comment]
...hmmm:  Dann soll PABS für die ambulanten Fälle keine Files mehr exportieren.  Ich möchte wirklich nicht noch "Spezialfunktionen" (check if `A` => Nothing & `S` => Import) in eine Schnittstelle einbauen, welche ja schon bald ersetzt wird. Zudem soll Tom sich der HL7 Schnittstelle widmen sowie weiteren Themen.  Ich denke diese Fälle wegzulassen ist für PABS ohne "Sonderprogrammierung" möglich.

**Comment by schoenenb** [comment]
Ja, das wäre eine Lösung. Wird aber spätestens mit dem ERP-Wechsel wieder nicht mehr funktionieren, wenn die Schnittstelle zum KIS dupliziert wird...
Eigentlich hatte ich ja das Issue gesucht, wo erklärt ist, weshalb nicht die tatsächlichen Daten eingefügt werden, sondern stets das Datum 1.1.1970. Das verstehe ich nämlich nicht (mehr). Wenn du noch weisst, wo das festgehalten ist, gerne mit dem Link bedienen.

**Comment by ottigerb** [comment]
Hmmm...    1.1.1970 ist ein UNIX "Standard-Datum" für 0. Am ersten Januar beginnt sozusagen die Unixzeit bei 0 Sekunden, was einen Hexadezimalwert von 00 00 00 00 entspricht! Dies war der Start von Unix 6 und alle weiteren Zeiten werden sozusagen mittels Sekunden addiert.

Ich erinnere mich noch daran, dass ich plötzlich erstaunt war, dass wir nun plötzlich so viele Fälle hatten. Anyway:  Mit der Einführung der neuen ERP-HL7 Schnittstelle gilt es doch festzulegen welche Fälle exportiert werden sollen. Nicht? Die können doch nicht einfach blind "übernehmen".
